combatEngine

```diff
@@ -60,22 +61,51 @@ function addFighter(type,team){
 };
 
 function getInitiative(){//Helper Function to create an initiative order.
+    var initiativeOrder = [];
     var needInitiative = fighters
-    var iterate = 0    
-    while (needInitiative.length > 0){}
+    //Finds the highest initiative of the unassigned units and moves it to the initiativeOrder array, repeats until no more need Initiative, then returns that array. Ties go to the unit earlier in the list.
+    while(needInitiative.length > 0){
+        if(wantLogs = true){
+            console.log("need Initiative is length " + needInitiative.length + " and contains: ")
+            console.log (needInitiative)
+        };
+        topInit = -999
+        for(const i in needInitiative){
+            if(wantLogs == true){
+                console.log("i is " + i);
+                console.log("current i's speed is: " + needInitiative[i].speed);
+            };
+            if (needInitiative[i].speed > topInit){
+                topInit = fighters[i].speed
+                topInitIndex = i
+            };
+        };
+        initiativeOrder.push(needInitiative.splice(topInitIndex,1)) 
```

Comment:
Long term replace this with a sort, but maybe only after you have a more complex combat object.

Reply:
Yea, I figure that the vast majority of the infrastructure here is doomed to start so this is just a quick way to count to like....20-30 combat objects with initiative and get a rough sort in a way that doesn't break